pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -194,6 +200,23 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		return sb.Unbind()
 	}
 
+	var emptyApplication v1alpha1.ApplicationSelector
+	if sbr.Spec.ApplicationSelector == emptyApplication ||
+		(sbr.Spec.ApplicationSelector.ResourceRef == "" &&
+			sbr.Spec.ApplicationSelector.LabelSelector.MatchLabels == nil) {
+		conditionsv1.SetStatusCondition(&sbr.Status.Conditions, conditionsv1.Condition{
```

Comment:
for emptyApplicationSelector CollectionReady is true, so there would be two updates, InjectionReady false and collectionReady True

Reply:
Resolved 03d1542